### PR TITLE
Weather helper

### PR DESCRIPTION
**Pull request by pgrawehr**
Superseedes #783 

Changes with respect to that PR:
- Updated some StyleCop issues
- Provided new free sources for formulas (Wikipedia and NOAA)
- Slightly adapted the formulas according to those sources
- Added more precise pressure reduction formula
- Added comparison and formatting members to Temperature and Pressure units

I did remove the "summer simmer index" calculation, as I could not find a reliable source for that one. A web search returned only non-scientific pages. I suggest to add the wind chill temperature instead.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,13 @@
 
 This repository contains the [System.Device.Gpio](https://www.nuget.org/packages/System.Device.Gpio) library and implementations for various boards like [Raspberry Pi](https://www.raspberrypi.org/) and [Hummingboard](https://www.solid-run.com/nxp-family/hummingboard/).
 
-The repository also contains [Iot.Device.Bindings](https://www.nuget.org/packages/Iot.Device.Bindings), a growing set of community-maintained [device bindings](src/devices/README.md) for IoT components.
+The repository also contains [IoT.Device.Bindings](https://www.nuget.org/packages/Iot.Device.Bindings), a growing set of community-maintained [device bindings](src/devices/README.md) for IoT components.
```
Comment (joperezr) [inline README.md:9]: This casing currently matches the name of the package so I would undo this change and the one bellow as well.

Comment (pgrawehr) [reply README.md]: True, this was probably an incorrectly resolved merge conflict.

**Comment by joperezr** [comment]
cc: @MarkCiliaVincenti as FYI since he did most of the work. @pgrawehr did you try pushing changes directly to @MarkCiliaVincenti PR?

File: src/devices/Common/tests/Common.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (krwq) [inline src/devices/Common/tests/Common.Tests.csproj:1]: cc: @joperezr can you validate this is correct?

File: src/devices/Dhtxx/samples/DhtSensor.sample.cs [hunk 3]
```diff
@@ -13,22 +14,29 @@ internal class Program
     {
         public static void Main(string[] args)
         {
-            Console.WriteLine("Hello DHT!");
+        Console.WriteLine("Hello DHT!");
```
Comment (krwq) [inline src/devices/Dhtxx/samples/DhtSensor.sample.cs:17]: indentation is off

File: src/devices/Units/tests/SetCultureForTest.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
```
Comment (krwq) [inline src/devices/Units/tests/SetCultureForTest.cs:1]: copyright header missing

File: src/devices/Units/tests/SetCultureForTest.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        private CultureInfo _previousCulture;
+        private CultureInfo _previousUiCulture;
+
+        public SetCultureForTest(string cultureName)
+        {
+            _previousCulture = CultureInfo.CurrentCulture;
+            _previousUiCulture = CultureInfo.CurrentUICulture;
+            var culture = new CultureInfo(cultureName, false);
+            Thread.CurrentThread.CurrentCulture = culture;
```
Comment (krwq) [inline src/devices/Units/tests/SetCultureForTest.cs:19]: while these will work for this project because we disabled parallelization I'd prefer we removed these tests or only tested it with explicit culture - this will be a time bomb - will start blowing when we start running tests in parallel (one way to do it reliably is to use RemoteExecutor, i.e.: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.Process/tests/ProcessTestBase.cs#L55)

File: src/devices/Units/tests/PressureTests.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
```
Comment (krwq) [inline src/devices/Units/tests/PressureTests.cs:1]: copyright header

**Comment by krwq** [comment]
LGTM after fixing minor comments and @joperezr sign-off on the csproj file. Thanks @MarkCiliaVincenti and @pgrawehr!

**Comment by MarkCiliaVincenti** [comment]
What, this will actually get merged at last? Had lost hope! :)

File: src/devices/Units/tests/SetCultureForTest.cs (hunk 5 above)
Comment (pgrawehr) [reply src/devices/Units/tests/SetCultureForTest.cs:19]: I don't think parallelization will cause problems here, since the culture is per Thread. Even if parallelization is active, no two tests will run simultaneously on the same thread.

**Comment by pgrawehr** [comment]
> 
> 
> cc: @MarkCiliaVincenti as FYI since he did most of the work. @pgrawehr did you try pushing changes directly to @MarkCiliaVincenti PR?

By default, it's not possible to push to somebody else's repository, this is also true for forks. I could have asked @MarkCiliaVincenti for permission, but I wasn't sure he would react.

**Comment by MarkCiliaVincenti** [comment]
To be honest I would like the credit to my work, especially at a time when I'm laid off due to Coronavirus and need to spice up my CV :)

**Comment by joperezr** [comment]
You would still get “credit” in the sense that he cherry picked your commits so they would still show you as author, but that said I agree with you that your PR deserves to be the one that gets merged, specially because it wasn’t really blocked by you, but instead blocked by us checking with legal about the source of those formulas. We can’t merge your PR for that same reason even if it is temporary but can you change your branch settings in your fork so you allow @pgrawehr to push to it so he would also get the credit for his work? I believe all you have to do is add him as a contributor but I’m not 100% sure. Then he’ll be able to push to your branch his fixes and we can go ahead and merge.

I’m really sorry for this whole situation guys, I totally take the blame here.

**Comment by MarkCiliaVincenti** [comment]
> You would still get “credit” in the sense that he cherry picked your commits so they would still show you as author, but that said I agree with you that your PR deserves to be the one that gets merged, specially because it wasn’t really blocked by you, but instead blocked by us checking with legal about the source of those formulas. We can’t merge your PR for that same reason even if it is temporary but can you change your branch settings in your fork so you allow @pgrawehr to push to it so he would also get the credit for his work? I believe all you have to do is add him as a contributor but I’m not 100% sure. Then he’ll be able to push to your branch his fixes and we can go ahead and merge.
> 
> I’m really sorry for this whole situation guys, I totally take the blame here.

Cool! I've invited @pgrawehr as a contributor on my branch.

**Comment by pgrawehr** [comment]
@MarkCiliaVincenti Not sure what you did, but it didn't do the trick... I think you need to go to the repository settings of your fork and add me as a contributor. joperezr is trying to work around it, as he should have access to your fork.

**Comment by MarkCiliaVincenti** [comment]
> @MarkCiliaVincenti Not sure what you did, but it didn't do the trick... I think you need to go to the repository settings of your fork and add me as a contributor. joperezr is trying to work around it, as he should have access to your fork.

That's exactly what I did but it says it's awaiting your response.

image

**Comment by joperezr** [comment]
Closing this one since the original PR is now merged

**Comment by pgrawehr** [comment]
> 
> 
> > @MarkCiliaVincenti Not sure what you did, but it didn't do the trick... I think you need to go to the repository settings of your fork and add me as a contributor. joperezr is trying to work around it, as he should have access to your fork.
> 
> That's exactly what I did but it says it's awaiting your response.
> 
> image

That's weird... I now found an E-Mail with the invitation, but when I click the link there, it says the invitation couldn't be found. But never mind, it's merged now ;-)

**Comment by MarkCiliaVincenti** [comment]
Yep sorry, by which time I deleted the fork and started afresh. Would have worked had I didn't.

**Comment by MarkCiliaVincenti** [comment]
By the way, thanks for your work, @pgrawehr, without your work it would probably never have been merged!
